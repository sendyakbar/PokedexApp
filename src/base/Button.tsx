import React from 'react';
import {
  StyleSheet,
  TouchableOpacity,
  Text,
  ViewStyle,
  TextStyle,
} from 'react-native';

type Props = {
  onPress: () => void;
  title: string;
};

export default function Button(props: Props): JSX.Element {
  const {onPress, title} = props;

  return (
    <TouchableOpacity
      style={styles.container}
      activeOpacity={0.8}
      onPress={onPress}>
      <Text style={styles.titleText}>{title}</Text>
    </TouchableOpacity>
  );
}

type StyleType = {
  container: ViewStyle;
  titleText: TextStyle;
};

const styles = StyleSheet.create<StyleType>({
  container: {
    padding: 10,
    borderWidth: 1,
    borderColor: 'grey',
    borderRadius: 4,
  },
  titleText: {
    fontSize: 12,
    color: 'grey',
  },
});
