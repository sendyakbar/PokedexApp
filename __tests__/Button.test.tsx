import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';

import Button from '../src/base/Button';

it('renders correctly', () => {
  const tree = renderer
    .create(<Button title="Title" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
