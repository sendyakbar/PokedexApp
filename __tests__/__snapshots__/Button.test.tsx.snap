// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "borderColor": "grey",
      "borderRadius": 4,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "grey",
        "fontSize": 12,
      }
    }
  >
    Title
  </Text>
</View>
`;
